synfig-core/src/synfig/rendering/opengl/internal/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+target_sources(libsynfig
+    PRIVATE
+		"${CMAKE_CURRENT_LIST_DIR}/antialiasing.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/buffers.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/context.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/environment.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/framebuffers.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/misc.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/predeclarations.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/samplers.cpp"
+		"${CMAKE_CURRENT_LIST_DIR}/shaders.cpp"
+)
+
+find_package(OpenGL REQUIRED)
+target_link_libraries(libsynfig PRIVATE OpenGL::GL OpenGL::GLU)
+
+set(RENDERING_OPENGL_INTERNAL_GLSL_FILES
+	"${CMAKE_CURRENT_LIST_DIR}/glsl/antialiased_textured_rect_fragment.glsl"
+	"${CMAKE_CURRENT_LIST_DIR}/glsl/antialiased_textured_rect_vertex.glsl"
+	"${CMAKE_CURRENT_LIST_DIR}/glsl/blend_fragment.glsl"
+	"${CMAKE_CURRENT_LIST_DIR}/glsl/color_fragment.glsl"
+	"${CMAKE_CURRENT_LIST_DIR}/glsl/simple_vertex.glsl"
+	"${CMAKE_CURRENT_LIST_DIR}/glsl/texture_fragment.glsl"
+	"${CMAKE_CURRENT_LIST_DIR}/glsl/texture_vertex.glsl"
+)
+
+add_custom_command(TARGET libsynfig POST_BUILD
```

Comment:
In this case, it will not update the shader if the shader is changed but the executable is not, because it only runs after the build is done. If executable sources not changed, then CMake doesn't rebuild it.

This is how this can be improved:
```CMake

set(RENDERING_OPENGL_INTERNAL_GLSL_FILES
	"glsl/antialiased_textured_rect_fragment.glsl"
	"glsl/antialiased_textured_rect_vertex.glsl"
	"glsl/blend_fragment.glsl"
	"glsl/color_fragment.glsl"
	"glsl/simple_vertex.glsl"
	"glsl/texture_fragment.glsl"
	"glsl/texture_vertex.glsl"
)

foreach (SHADER IN ITEMS ${RENDERING_OPENGL_INTERNAL_GLSL_FILES})
	configure_file("${CMAKE_CURRENT_LIST_DIR}/${SHADER}" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${SHADER}" COPYONLY)
endforeach ()
```

`configure_file` updates files if they have changed. Also directory will be created automatically.

P.S. Also don't forget to add `install` command.

Reply:
Okay, I will update this.